### PR TITLE
List of clients

### PR DESCRIPTION
**Pull request by Francois389**
Add clients list with API call.

### CODE REVIEW

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/Home.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import com.google.android.material.tabs.TabLayout;
+import com.google.android.material.tabs.TabLayoutMediator;
+
+import fr.iut_rodez.pathpilot_android_client.R;
+import fr.iut_rodez.pathpilot_android_client.home.clients.FragmentClients;
+import fr.iut_rodez.pathpilot_android_client.login.LoginService;
+import fr.iut_rodez.pathpilot_android_client.login.TokenJWT;
+
+public class Home extends AppCompatActivity {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/Home.java:21]: class Javadoc comments to add

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import fr.iut_rodez.pathpilot_android_client.model.Client.ClientArrayAdapter;
+
+public class ClientService {
+
+    public static final String API_BASE_URL = BuildConfig.API_BASE_URL + "api/clients";
+    private static final String TAG = ClientService.class.getSimpleName();
+
+    private static RequestQueue requestQueue;
+
+    public static void getClients(Context context, ListView listClientsView) {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java:36]: Add singature

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/FragmentClients.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import androidx.fragment.app.Fragment;
+
+import java.util.List;
+
+import fr.iut_rodez.pathpilot_android_client.R;
+import fr.iut_rodez.pathpilot_android_client.home.Home;
+import fr.iut_rodez.pathpilot_android_client.model.Client;
+
+public class FragmentClients extends Fragment {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/FragmentClients.java:20]: Might be good to add class javadoc comment

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/login/TokenJWT.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package fr.iut_rodez.pathpilot_android_client.login;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import androidx.annotation.NonNull;
+
+import java.util.Date;
+
+public class TokenJWT implements Parcelable {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/login/TokenJWT.java:10]: Change to JWTToken to have english name and javadoc comments for the class

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/model/Client.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+import androidx.annotation.Nullable;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import fr.iut_rodez.pathpilot_android_client.R;
+
+public class Client implements Parcelable {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/model/Client.java:22]: Add Javadoc class comments

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java [hunk 6]
```diff
@@ -1,16 +1,24 @@
...
 import androidx.activity.EdgeToEdge;
 import androidx.appcompat.app.AppCompatActivity;
 
-public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+import fr.iut_rodez.pathpilot_android_client.R;
+import fr.iut_rodez.pathpilot_android_client.login.Login;
+import fr.iut_rodez.pathpilot_android_client.signup.SignUpService.SignUpInput;
+import fr.iut_rodez.pathpilot_android_client.util.Popup;
+
+public class SignUp extends AppCompatActivity {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java:17]: Add Javadoc class comments

File: app/src/main/res/drawable/icon_car.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
...5,11.44L7.93,31.19C3.28,32.98 0,37.36 0,42.5V63.75C0,67.11 2.79,69.82 6.25,69.82H12.75C14.24,78.44 21.95,85 31.25,85C40.54,85 48.25,78.44 49.74,69.82H75.26C76.75,78.44 84.46,85 93.75,85C103.05,85 110.76,78.44 112.25,69.82H118.75C122.21,69.82 125,67.11 125,63.75V54.64C125,42.27 115.47,32.06 103.17,30.55H103.15ZM84.89,63.75C85.29,62.61 85.91,61.55 86.74,60.65C87.56,59.74 88.55,59 89.67,58.48C90.79,57.95 92,57.64 93.24,57.57C94.48,57.51 95.72,57.68 96.89,58.08C98.06,58.48 99.14,59.11 100.06,59.92C100.98,60.73 101.73,61.71 102.25,62.8C102.78,63.89 103.08,65.08 103.13,66.28C103.19,67.49 102.99,68.69 102.56,69.82C102.16,70.96 101.54,72.02 100.72,72.92C99.89,73.83 98.9,74.57 97.78,75.09C96.66,75.62 95.45,75.93 94.21,76C92.97,76.06 91.73,75.89 90.56,75.49C89.39,75.09 88.31,74.46 87.39,73.65C86.47,72.84 85.72,71.86 85.2,70.77C84.67,69.68 84.37,68.5 84.32,67.29C84.26,66.09 84.46,64.88 84.89,63.75ZM31.25,57.68C33.73,57.68 36.12,58.64 37.87,60.35C39.63,62.05 40.62,64.37 40.62,66.79C40.62,69.2 39.63,71.52 37.87,73.23C36.12,74.93 33.73,75.89 31.25,75.89C28.76,75.89 26.37,74.93 24.62,73.23C22.86,71.52 21.87,69.2 21.87,66.79C21.87,64.37 22.86,62.05 24.62,60.35C26.37,58.64 28.76,57.68 31.25,57.68Z"
```
Comment (Odonata971) [inline app/src/main/res/drawable/icon_car.xml:7]: Error line 7:286 lol

**Review by Odonata971 — CHANGES_REQUESTED** [review]

File: app/src/main/res/drawable/icon_car.xml (hunk 7 above)
Comment (Francois389) [reply app/src/main/res/drawable/icon_car.xml:7]: 🥇

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java (hunk 6 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/model/Client.java (hunk 5 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/model/Client.java:25]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/login/TokenJWT.java (hunk 4 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/login/TokenJWT.java]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/FragmentClients.java (hunk 3 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/FragmentClients.java:21]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java (hunk 2 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java:38]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/Home.java (hunk 1 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/Home.java:25]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java (hunk 6 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/f495a2fcc43e0fc3d934ae46d71b9af6a0856994

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java (hunk 2 above)
Comment (Odonata971) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java:38]: not on the method, there is supposed to be a signature there

**Review by Odonata971 — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java (hunk 2 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java:46]: Really fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/44/commits/1828c9ee2511dbcbee079ec147a547bac356039b

**Review by Odonata971 — APPROVED** [review]
LGTM

**Pull request merged**